---
title: C26165
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C26165
helpviewer_keywords:
- C26165
ms.assetid: a1d89bd6-08f3-4215-8a0c-b8ecfeb0cffc
ms.openlocfilehash: 6868c3296f601173e6bc11fe0f1fcc2b2538311f
ms.sourcegitcommit: c1fd917a8c06c6504f66f66315ff352d0c046700
ms.translationtype: MT
ms.contentlocale: ru-RU
ms.lasthandoff: 09/16/2020
ms.locfileid: "90683940"
---
# <a name="c26165"></a>C26165

> предупреждение C26165: возможно, не удалось освободить блокировку \<lock> в функции \<func> .

Предупреждение C26165 напоминает предупреждение [C26115](../code-quality/c26115.md) , за исключением того, что уровень достоверности меньше. Например, функция может содержать ошибки аннотации.

## <a name="examples"></a>Примеры

Следующий код создает предупреждение C26165.

```cpp

_Create_lock_level_(LockLevelOne);
_Create_lock_level_(LockLevelTwo);

struct LockLevelledStruct
{
    _Has_lock_level_(LockLevelOne) CRITICAL_SECTION a;
    _Has_lock_level_(LockLevelTwo) CRITICAL_SECTION b;
};

_Lock_level_order_(LockLevelOne, LockLevelTwo);

_Acquires_lock_(s->b) void GetLockFunc(LockLevelledStruct* s)
{
    EnterCriticalSection(&s->b);
}

void testLockLevelledStruct(LockLevelledStruct* s) // Warning C26165
{
    EnterCriticalSection(&s->a);
    GetLockFunc(s);
    LeaveCriticalSection(&s->a);
}
```

Чтобы исправить это предупреждение, измените предыдущий пример на следующий.

```cpp

_Create_lock_level_(LockLevelOne);
_Create_lock_level_(LockLevelTwo);

struct LockLevelledStruct
{
    _Has_lock_level_(LockLevelOne) CRITICAL_SECTION a;
    _Has_lock_level_(LockLevelTwo) CRITICAL_SECTION b;
};

_Lock_level_order_(LockLevelOne, LockLevelTwo);

_Acquires_lock_(s->b) void GetLockFunc(LockLevelledStruct* s)
{
    EnterCriticalSection(&s->b);
}

_Releases_lock_(s->b) void ReleaseLockFunc(LockLevelledStruct* s)
{
    LeaveCriticalSection(&s->b);
}

void testLockLevelledStruct(LockLevelledStruct* s) // OK
{
    EnterCriticalSection(&s->a);
    GetLockFunc(s);
    ReleaseLockFunc(s);
    LeaveCriticalSection(&s->a);
}
```
